firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalStore.java

```diff
@@ -329,50 +329,35 @@ public SnapshotVersion getLastRemoteSnapshotVersion() {
     return persistence.runTransaction(
         "Apply remote event",
         () -> {
+          SnapshotVersion remoteVersion = remoteEvent.getSnapshotVersion();
```

Comment:
Question: this variable is now much further from its first point of use, and I didn't notice any references to it in-between. Perhaps move it back to the end of this function?

Reply:
It's used in line 354. I moved it to the top of the function body - let me know if you think that's better. I can otherwise also inline the usage in line 354.